### PR TITLE
[WIP] Add initial report usage files

### PR DESCRIPTION
**Pull request by klapko-stripe**
## Summary

- Adds files for reporting usage for Go and .NET

### CODE REVIEW

File: usage-based-subscriptions/server/dotnet/Controllers/ReportUsage.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+using System;
```
Comment (klapko-stripe) [inline usage-based-subscriptions/server/dotnet/Controllers/ReportUsage.cs:1]: Lines 1-35 I've added but I have no idea if they're right. Lines 37+ were copy and pasted.

File: usage-based-subscriptions/server/go/report_usage.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package main
```
Comment (klapko-stripe) [inline usage-based-subscriptions/server/go/report_usage.go:1]: I'm not sure what the `imports` should look like for this. Lines 10+ were copy and pasted.

File: usage-based-subscriptions/server/dotnet/ReportUsage.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            // https://stripe.com/docs/api/subscriptions/object#subscription_object-items.
+            var subscriptionItemId = "{{SUBSCRIPTION_ITEM_ID}}";
+
+            // The usage number you've been keeping track of in your database for the last 24 hours.
+            var usageQuantity = 100L;
+
+            // The idempotency key allows you to retry this usage record call if it fails.
+            var idempotencyKey = System.Guid.NewGuid().ToString();
+
+            var timestamp = DateTimeOffset.FromUnixTimeSeconds(1597408025).UtcDateTime;
```
Comment (suz-stripe) [inline usage-based-subscriptions/server/dotnet/ReportUsage.cs:32]: should this timestamp perhaps be switched to getting the current time?

File: usage-based-subscriptions/server/go/report_usage.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  // https://stripe.com/docs/api/subscriptions/object#subscription_object-items.
+  var subscriptionItemId = "{{SUBSCRIPTION_ITEM_ID}}"
+
+  // The usage number you've been keeping track of in your database for the last 24 hours.
+  usageQuantity := int64(100)
+
+  params := &stripe.UsageRecordParams{
+    SubscriptionItem: stripe.String(subscriptionItemId),
+    Quantity: stripe.Int64(usageQuantity),
+    Timestamp: stripe.Int64(1597415390),
```
Comment (suz-stripe) [inline usage-based-subscriptions/server/go/report_usage.go:30]: should this timestamp perhaps be switched to getting the current time?

File: usage-based-subscriptions/server/go/report_usage.go (hunk 2 above)
Comment (suz-stripe) [reply usage-based-subscriptions/server/go/report_usage.go:1]: I think this looks okay to me!

**Review by suz-stripe — APPROVED** [review]
I didn't manually run these files but I assume y'all did anyway 😄 

If still in doubt, run this briefly by Billing to get a second opinion but overall these look ok

**Pull request merged**